### ISSUE TITLE
ZeroDivisionError: float division by zero

### ISSUE THREAD

**Issue by visuallization** [suragnair/alpha-zero-general#288]
Hey there,

First of all thank you for sharing this repo with the public. Really cool project!

Rearding the opened issue: I experience currently a problem with MCTS in `def getActionProb`, namely the error `ZeroDivisionError: float division by zero`. It originates from `probs = [x / counts_sum for x in counts]` and the problem is that all the elements in `counts` are 0 because the state/action pair has not been discovered and saved in self.Nsa yet (`counts = [self.Nsa[(s, a)] if (s, a) in self.Nsa else 0 for a in range(self.game.getActionSize())]`). I am not sure how to handle this. I am currently trying to make the game of hex work ind the project.

Cheers

**Comment by goshawk22** [comment]
#191 might help

**Comment by visuallization** [comment]
Yes, thanks for pointing this out. However I already dicovered this one and I couldn't find anything similiar in my code. I made sure that I always copy the pieces in the game code. It just seems that mcts is not exploring edges which the players in the arena then want to play. But I guess it should only play discovered moves (s/a) in the arena since it uses greedy search, right?

**Comment by visuallization** [comment]
> Could you share your code?

I gladly do: https://github.com/visuallization/alpha-zero-general-hex/tree/master/hex
The relevant files are `HexGame.py` and `HexBoard.py`. Furthermore it currently uses the NeuralNet architecture from Othello.

**Comment by visuallization** [comment]
I am not working with pass though, like Othello does because there is no pass in Hex.

**Comment by visuallization** [comment]
Okay there might be an issue how I represent the `canonicalBoard` in `getCanonicalForm`. If I just return the board without inverting it, the issue of not finding the state in self.Nsa does not arise anymore.

```
def getCanonicalForm(self, positions, player):
        board = HexBoard(self.size)
        board.positions = np.copy(positions)
        return board.positions
        #return player * board.positions
```

**Comment by jamesbraza** [comment]
@visuallization did you ever figure this out?

I believe from looking at your fork that this issue was resolved, and is a dupe of https://github.com/suragnair/alpha-zero-general/issues/191
